_posts/2025-01-13-mcp-server.adoc

```diff
@@ -0,0 +1,263 @@
+    return forecast.properties().periods().stream().map(period -> {
+            // Template for each forecast period
+            return Qute.fmt(
+                """
+                        Temperature: {p.temperature}°{p.temperatureUnit}
+                        Wind: {p.windSpeed} {p.windDirection}
+                        Forecast: {p.detailedForecast}
+                        """,
+                Map.of("p", period)).toString();
+        }).collect(Collectors.joining("\n---\n"));
+    }
+----
+
+=== Implementing MCP Tools
+
+Now let's implement the actual MCP tools. The `@Tool` annotation from `io.quarkiverse.mcp.server` marks methods as available tools, while `@ToolArg` describes the parameters:
+
+[source,java]
+----
+@Tool(description = "Get weather alerts for a US state.")
+String getAlerts(@ToolArg(description = "Two-letter US state code (e.g. CA, NY)") String state) {
+    return formatAlerts(weatherClient.getAlerts(state));
+}
+
+@Tool(description = "Get weather forecast for a location.")
+String getForecast(
+    @ToolArg(description = "Latitude of the location") double latitude,
+    @ToolArg(description = "Longitude of the location") double longitude) {
+    
+    // First get the point metadata which contains the forecast URL
+    var points = weatherClient.getPoints(latitude, longitude);
+    // Extract the forecast URL using Qute template
+    var url = Qute.fmt("{p.properties.forecast}", Map.of("p", points));
+    // Get and format the forecast
+    return formatForecast(weatherClient.getForecast(url));
+}
+----
+
+[NOTE]
+====
+The forecast API requires a two-step process where we first get point metadata and then use a URL from that response to fetch the actual forecast.
+====
+
+== Running the Server
+
+To simplify deployment and development, we'll package the server as an uber-jar and enable file logging (since stdio is reserved for the MCP protocol):
+
+[source,properties]
+----
+quarkus.package.uber-jar=true
+quarkus.log.file.enable=true
+quarkus.log.file.path=weather-quarkus.log
+----
+
+After running `mvn install`, you can use JBang to run the server using its Maven coordinates: `org.acme:weather:1.0.0-SNAPSHOT:runner`
+or manually using `java -jar target/weather-1.0.0-SNAPSHOT-runner.jar`.
+
+=== Integration with Claude
+
+Add this to your `claude_desktop_config.json`:
+
+[source,json]
+----
+{
+    "mcpServers": {
+        "weather": {
+            "command": "jbang",
+            "args": ["--quiet", 
+                    "org.acme:weather:1.0.0-SNAPSHOT:runner"]
+        }
+    }
+}
+----
+
+The `--quiet` flag prevents JBang's output from interfering with the MCP protocol.
+
+image::claude-tools.png[Claude Tools Integration]
+
+[NOTE]
+====
+You can also run the server directly withou using java - then it would be something like `java -jar <FULL PATH>/weather-1.0.0-SNAPSHOT-runner.jar`. We use JBang here because simpler if you want to share with someone who does not want to build the MCP server locally.
```

Comment:
```suggestion
You can also run the server directly without using java - then it would be something like `java -jar <FULL PATH>/weather-1.0.0-SNAPSHOT-runner.jar`. We use JBang here because simpler if you want to share with someone who does not want to build the MCP server locally.
```